x-pack/legacy/plugins/siem/public/components/timeline/fetch_kql_timeline.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { memo, useEffect } from 'react';
+import { connect } from 'react-redux';
+import { ActionCreator } from 'typescript-fsa';
+import { StaticIndexPattern } from 'ui/index_patterns';
+
+import { inputsModel, KueryFilterQuery, timelineSelectors, State } from '../../store';
+import { inputsActions } from '../../store/actions';
+import { InputsModelId } from '../../store/inputs/constants';
+import { useUpdateKql } from '../../utils/kql/use_update_kql';
+
+interface TimelineKqlFetchRedux {
+  kueryFilterQuery: KueryFilterQuery | null;
+  kueryFilterQueryDraft: KueryFilterQuery | null;
+}
+
+interface TimelineKqlFetchDispatch {
+  setTimelineQuery: ActionCreator<{
+    id: string;
+    inputId: InputsModelId;
+    inspect: inputsModel.InspectQuery | null;
+    loading: boolean;
+    refetch: inputsModel.Refetch | inputsModel.RefetchKql | null;
+  }>;
+}
+
+export interface TimelineKqlFetchProps {
+  id: string;
+  indexPattern: StaticIndexPattern;
+  inputId: InputsModelId;
+}
+
+type OwnProps = TimelineKqlFetchProps & TimelineKqlFetchRedux & TimelineKqlFetchDispatch;
+
+const TimelineKqlFetchComponent = memo<OwnProps>(
+  ({ id, indexPattern, inputId, kueryFilterQuery, kueryFilterQueryDraft, setTimelineQuery }) => {
+    useEffect(() => {
+      setTimelineQuery({
+        id: 'kql',
+        inputId,
+        inspect: null,
+        loading: false,
+        refetch: useUpdateKql({
+          indexPattern,
+          kueryFilterQuery,
+          kueryFilterQueryDraft,
+          storeType: 'timelineType',
+          type: null,
+          timelineId: id,
+        }),
+      });
+    }, [kueryFilterQueryDraft, kueryFilterQuery, id]);
```

Comment:
optional: should the 2nd arg to `useEffect` this also include `indexPattern`, in case we support dynamic updates to it in the future?

Reply:
you are right but I feel like it is a really big object to compare, I think we will might have to figure out a better way than compare an object with an object